src/main/scala/com/kinja/amqp/AmqpClientFactory.scala

```diff
@@ -15,25 +15,23 @@ class AmqpClientFactory {
 		actorSystem: ActorSystem,
 		logger: Logger,
 		ec: ExecutionContext,
-		getWriteConnection: () => Connection,
-		getReadConnecion: () => Connection,
-		hostname: String
+		messageStores: Map[String, MessageStore]
 	): AmqpClientInterface =
 		{
 			if (config.testMode) {
 				new NullAmqpClient
 			} else {
 				val connection: ActorRef = createConnection(config, actorSystem)
-				val messageStore: MessageStore = createMessageStore(
-					config, actorSystem, logger, ec, getWriteConnection, getReadConnecion, hostname
+				val bufferedMessageStores = messageStores.mapValues(createMessageStore(
```

Comment:
Never use `mapValues`!!! That should be removed from the Scala library. It only creates a view of the original map with the function applied to items every single f*ing time they're accessed. So here `createMessageStore` will be called each and every time you access a value from `bufferedMessageStores`.

Reply:
Oh, shit, I wasn't aware of that. It's a landmine indeed. Thanks!

Reply:
Done. Removed `mapValues` in several other places, just to be on the safe side.

Is there a wart for that?